src/main/java/io/github/syst3ms/skriptparser/effects/EffCase.java

```diff
@@ -0,0 +1,124 @@
+package io.github.syst3ms.skriptparser.effects;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Effect;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.Statement;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.log.ErrorType;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.parsing.SyntaxParser;
+import io.github.syst3ms.skriptparser.sections.SecSwitch;
+import io.github.syst3ms.skriptparser.types.comparisons.Comparators;
+import io.github.syst3ms.skriptparser.types.comparisons.Relation;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+/**
+ * This effect is written underneath the {@link SecSwitch switch} section to match the given expression.
+ * The statement inside this effect will only be executed if it matches the given expression.
+ * One may use 'or'-lists to match multiple expressions at once.
+ * The default part can be used to provide actions when no match was found.
+ * Note that you can only use one default statement and that equivalent matches are prohibited.
+ *
+ * @name Case
+ * @type EFFECT
+ * @pattern (case|when) %*objects%[,] [do] <.+>
+ * @pattern ([by] default|otherwise)[,] [do] <.+>
+ * @since ALPHA
+ * @author Mwexim
+ * @see SecSwitch
+ */
+public class EffCase extends Effect {
```

Comment:
I'm against inlined effects like these.

Reply:
Agree, we should make them as a section, not a inline effect